TrackingTools/TrackRefitter/src/TrackTransformer.cc

```diff
@@ -34,12 +36,24 @@ TrackTransformer::TrackTransformer(const ParameterSet& parameterSet):
     theSmootherName(parameterSet.getParameter<string>("Smoother")),  
     thePropagatorName(parameterSet.getParameter<string>("Propagator")),
     theTrackerRecHitBuilderName(parameterSet.getParameter<string>("TrackerRecHitBuilder")),
-    theMuonRecHitBuilderName(parameterSet.getParameter<string>("MuonRecHitBuilder"))
+    theMuonRecHitBuilderName(parameterSet.getParameter<string>("MuonRecHitBuilder")),
+    theMTDRecHitBuilderName(parameterSet.getParameter<string>("MTDRecHitBuilder"))
     {}
 
 /// Destructor
 TrackTransformer::~TrackTransformer(){}
 
+void TrackTransformer::fillPSetDescriptions(edm::ParameterSetDescription& desc) {
+  desc.add<bool>("DoPredictionsOnly",false);
+  desc.add<std::string>("Fitter",std::string("KFFitterForRefitInsideOut"));
+  desc.add<std::string>("Smoother",std::string("KFSmootherForRefitInsideOut"));
+  desc.add<std::string>("Propagator",std::string("SmartPropagatorAnyRK"));
+  desc.add<std::string>("RefitDirection",std::string("alongMomentum"));  
+  desc.add<bool>("RefitRPCHits",true);
+  desc.add<std::string>("TrackerRecHitBuilder",std::string("WithTrackAngle"));
+  desc.add<std::string>("MuonRecHitBuilder",std::string("MuonRecHitBuilder"));
+  desc.add<std::string>("MTDRecHitBuilder",std::string("MTDRecHitBuilder"));
```

Comment:
@Martin-Grunewald @lgray as far as I see this is the place where the dependency of HLT on the MTD code enters (through L3MuonProducer). What would be the suggestion? To make it dependent on Era definition?

Reply:
Is it correct to assume that the setting of the new parameter ```("MTDRecHitBuilder",std::string("MTDRecHitBuilder")``` which is not in the HLT config yet and thus uses the fillDescriptions default provided,  reproduces the old behaviour (without MTD)?
It should be set this way in fillDescriptions, with reco config files setting it to whatever you like for RECO.
No, HLT does not really use eras, as each menu is tied to a data taking period.

Reply:
Actually, why did you change the values of the other parameters?

Reply:
I was using the default value from a RECO configuration here. Since there was no fill descriptions for TrackTransformer before this,  then all previously existing values will be overridden by configuration parameters, and only MTDRecHitBuilder will be used as a default.

If the MTD tracking geometry is not available in the tracking geometry this parameter has no effect.